app/models/ems_refresh/save_inventory_helper.rb

```diff
@@ -42,6 +42,7 @@ def save_inventory_multi(association, hashes, deletes, find_key, child_keys = []
       deletes.reload
     end
     deletes = deletes.to_a
+    deletes_index = deletes.each_with_object({}) { |x, obj| obj[x] = x }
```

Comment:
Also, not sure about the name `indexed_vms`...seems too generic.  Maybe `vms_by_uid_ems`?